src/rgw/rgw_auth.cc

```diff
@@ -818,15 +818,7 @@ bool rgw::auth::RemoteApplier::is_owner_of(const rgw_owner& o) const
     return false;
   }
 
-  if (info.acct_user.tenant.empty()) {
-    const rgw_user tenanted_acct_user(info.acct_user.id, info.acct_user.id);
-
-    if (tenanted_acct_user == *uid) {
-      return true;
-    }
-  }
-
-  return info.acct_user == *uid;
+  return owner_acct_user == *uid;
```

Comment:
This seems to have gotten simplier - why is the old code more complicated, and does this correctly handle anonymous access (unauthenticated swift)?

Reply:
> why is the old code more complicated

i don't know why it did that, but it looked wrong to my naive eyes: if `load_acct_info()` selects a `uid$uid` user, that identity shouldn't also have ownership of resources created by a separate `$uid` user in the global tenant namespace

this change is probably not necessary to fix the upgrade regression, so i can remove it if you're concerned

> and does this correctly handle anonymous access (unauthenticated swift)?

i don't think swift anonymous access goes through RemoteApplier in the first place

Reply:
> this change is probably not necessary to fix the upgrade regression, so i can remove it if you're concerned

turns out that ThirdPartyAccountApplier needs to call this `is_owner_of()` function before calling `load_acct_info()`: 
https://github.com/ceph/ceph/blob/69149bb/src/rgw/rgw_auth_filters.h#L178-L181

i'll test again with this part reverted